src/main/kotlin/sideChain/iroha/IrohaConsumerImpl.kt

```diff
@@ -0,0 +1,153 @@
+package sideChain.iroha
+
+import notary.IrohaOrderedBatch
+import mu.KLogging
+import notary.Notary
+import main.Configs
+import notary.IrohaCommand
+import java.nio.file.Paths
+import java.math.BigInteger
+
+// iroha
+import ByteVector
+import ModelCrypto
+import ModelTransactionBuilder
+import ModelProtoTransaction
+import Keypair
+import PublicKey
+import kotlin.reflect.jvm.internal.impl.protobuf.InvalidProtocolBufferException
+import iroha.protocol.BlockOuterClass
+import iroha.protocol.CommandServiceGrpc
+
+import io.grpc.ManagedChannelBuilder
+
+/**
+ * Dummy implementation of [IrohaConsumer] with effective dependency
+ */
+class IrohaConsumerImpl(notary: Notary) : IrohaConsumer {
+
+    /** Iroha keypair */
+    val keypair: Keypair
+
+    init {
+        notary.irohaOutput().subscribe {
+            onIrohaEvent(it)
+        }
+        val crypto = ModelCrypto()
+        keypair = crypto.convertFromExisting(
+            readKeyFromFile(Configs.pubkeyPath),
+            readKeyFromFile(Configs.privkeyPath)
+        )
+    }
+
+    /**
+     * Read key from file specified by [path]
+     */
+    private fun readKeyFromFile(path: String): String {
+        try {
+            return String(java.nio.file.Files.readAllBytes(Paths.get(path)))
+        } catch (e: java.io.IOException) {
+            logger.error("Unable to read key files.\n $e")
+            System.exit(1)
+        }
+        return ""
+    }
+
+    /** Convert [ByteVector] to [ByteArray] */
+    private fun toByteArray(blob: ByteVector): ByteArray {
```

Comment:
This function looks like Util. Maybe remove it from consumer code and use a dependency?

Reply:
I'm confused with `signAndAddSignature()`. It returns custom type `ByteVector ` that should be converted to `ByteArray`. I don't think that function can be used elsewhere.